google_guest_agent/cfg/cfg_integ_test.go

```diff
@@ -0,0 +1,101 @@
+//  Copyright 2023 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+//go:build integration
+// +build integration
+
+package cfg
+
+import (
+	"os"
+	"runtime"
+	"testing"
+)
+
+func TestConfigDefault(t *testing.T) {
+	configFile := defaultConfigFile(runtime.GOOS)
+	testcases := []struct {
+		name           string
+		extraDefault   string
+		distroConfig   string
+		templateConfig string
+		userConfig     string
+		output         bool
+	}{
+		{
+			name:           "user config override",
+			extraDefault:   "[NetworkInterfaces]\nSetup = true\n",
+			distroConfig:   "[NetworkInterfaces]\nSetup = true\n",
+			templateConfig: "[NetworkInterfaces]\nSetup = true\n",
+			userConfig:     "[NetworkInterfaces]\nSetup = false\n",
+			output:         false,
+		},
+		{
+			name:           "template config override",
+			extraDefault:   "[NetworkInterfaces]\nSetup = true\n",
+			distroConfig:   "[NetworkInterfaces]\nSetup = true\n",
+			templateConfig: "[NetworkInterfaces]\nSetup = false\n",
+			userConfig:     "",
+			output:         false,
+		},
+		{
+			name:           "distro config override",
+			extraDefault:   "[NetworkInterfaces]\nSetup = true\n",
+			distroConfig:   "[NetworkInterfaces]\nSetup = false\n",
+			templateConfig: "",
+			userConfig:     "",
+			output:         false,
+		},
+		{
+			name:           "extra default override",
+			extraDefault:   "[NetworkInterfaces]\nSetup = false\n",
+			distroConfig:   "",
+			templateConfig: "",
+			userConfig:     "",
+			output:         false,
+		},
+		{
+			// If this fails, integration test results are not valid
+			name:           "default is true",
+			extraDefault:   "",
+			distroConfig:   "",
+			templateConfig: "",
+			userConfig:     "",
+			output:         true,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := os.WriteFile(configFile+".distro", []byte(tc.distroConfig), 0777)
```

Comment:
Please use testing.T.TempDir() to guarantee these files are being removed, also to avoid messing the environment please make defaultConfigFile() swappable like defaultDataSource() (and make defaultDataSource() to use the pointer) so we can swap it for the test case and make it return the right file name constructed based on the tests temp dir.